qtism/data/content/xhtml/ObjectElement.php

```diff
@@ -201,12 +201,12 @@ public function hasWidth()
      *
      * A negative value describes that no height is provided.
      *
-     * @param int $height A height.
+     * @param string|int $height A height.
      * @throws InvalidArgumentException If $height is not an integer value.
      */
     public function setHeight($height)
     {
-        if (is_int($height)) {
+        if ((is_int($height) && $height === -1) || Format::isXhtmlLength($height) === true) {
             $this->height = $height;
         } else {
```

Comment:
Bonus point: you can remove cognitive load by turning structure like
```
if (condition) {
  doSomething();
} else {
  throw nex Exception(...);
}
```
to
```
if (!condition) {
  throw nex Exception(...);
}
doSomething();
```

Reply:
I tried to refactor but it only increases cognitive load